detect_secrets/core/bidirectional_iterator.py

```diff
@@ -0,0 +1,31 @@
+class BidirectionalIterator(object):
+    def __init__(self, collection):
+        self.collection = collection
+        self.index = -1 #starts on -1, as index is increased _before_ getting result
+        self.step_back_once = False 
+
+    def __next__(self):
+        try:
+            if self.step_back_once:
+                self.index -= 1
+                self.step_back_once = False 
+                if self.index < 0:
+                    raise StopIteration
+            else:
+                self.index += 1
+            result = self.collection[self.index]
```

Comment:
Nit: Generally people recommend that only the code that throws the relevant exception should be inside of the `try: except:`, in this case `result = self.collection[self.index]` (This also cuts down on [`Hadouken code`](http://i.imgur.com/BtjZedW.jpg), as there will be 2 indentation levels instead of 3.)

```python
    def __next__(self):
        if self.step_back_once:
            self.index -= 1
            self.step_back_once = False 
            if self.index < 0:
                raise StopIteration
        else:
            self.index += 1

        try:
            result = self.collection[self.index]
        except IndexError:
            raise StopIteration

        return result
```

p.s. 'Nit' means nitpick, i.e. something that doesn't really matter that much.